bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java

```diff
@@ -188,12 +188,17 @@ public Collection<WorkItem> prRun(ScratchArea scratchArea) {
         if (nextCommand.isEmpty()) {
             log.info("No new non-external PR commands found, stopping further processing");
 
-            if (!bot.isAutoLabelled(pr) && !pr.isClosed()) {
-                // When all commands are processed, it's time to check labels
-                return List.of(new LabelerWorkItem(bot, prId, errorHandler, triggerUpdatedAt));
-            } else {
+            // If there is no label configuration, don't generate LabelerWorkItem
+            if (bot.labelConfiguration().allowed().isEmpty()) {
+                bot.setAutoLabelled(pr);
                 return List.of();
             }
+
+            if (!pr.isClosed()) {
+                return List.of(new LabelerWorkItem(bot, prId, errorHandler, triggerUpdatedAt));
```

Comment:
At this stage in PullRequestCommandWorkItem, we have already loaded all PR comments, so I think it would be worth checking if there is any need for scheduling a LabelerWorkItem. What I mean is, we can check if the head hash of the PR is different from the last labeler commit marker. This could be implemented in a static method on `LabelereWorkItem` to keep the details in one place.

Reply:
When I was working on it, I have the same thought with you, I added the check and did some tests, later I found there is a case which is that the user use the "/label" command to add the label, in this case, we need LabelerWorkItem to check whether the labels can be upgraded to group labels.

Reply:
Or there is another Option, we can add the check here. And in pullRequestWorkItem, after processing a label command, we can schedule a LabelerWorkItem

Reply:
Hm, that could work, but it would also mean that we need to schedule an initial LabelerWorkItem when bot is starting up, or rather, we need to let the very first one through regardless of commit hash. This is in case a LabelCommand was issued and handled right before bot was restarted. 

The alternative would be to compare dates on comments. If the label "command handled" comment has a later date than the last update of the auto labeler comment, then we need to run. If we do that, then the labeler needs to always touch the comment to mark when it was last run. Not sure which is best. If we end up spawning a lot of LabelereWorkItems, it could turn into a lot of comment updates.

Reply:
I think the first one is pretty easy, we just need to check if the pr has been marked as auto labeled or not.